### PR TITLE
feat(server):add versioned api structure

### LINKED ISSUE #39: Need NodeJS versioned API strcuture
Need to have a versioned API structure so that we can change versions of the API that are old.
For eg: https://example.com/api/v1/test_health

### PR DESCRIPTION
**Pull request by piyushagwl007**
FIxes: #39

### CODE REVIEW

File: server/package.json
Comment (piyushagwl007) [inline server/package.json:18]: The command `nodemon -r esm -r dotenv/config index.js -` will work like this:
1. require ESM (for using ES6 format in our server application)
2. require dotenv/config (for loading environment variables in the application)
3. Then start the index.js file

Note: Work of -r is preloading the specified module at startup.

**Review by nimish-gupta — APPROVED** [review]

File: server/src/configs/app.js
Comment (karansapolia) [inline server/src/configs/app.js:7]: Is this line feed needed?

File: server/src/configs/app.js
Comment (karansapolia) [inline server/src/configs/app.js:12]: Similarly this one?

File: server/src/controller/apis/user.js
Comment (karansapolia) [inline server/src/controller/apis/user.js:7]: Any specific reason why the route is set to '/health' ?

File: server/src/routes/index.js
Comment (karansapolia) [inline server/src/routes/index.js:7]: More of a learning question. What does return next() do here? :slightly_smiling_face:

**Review by karansapolia — APPROVED** [review]
Like that all parts are documented individually. @nihalanisumit has more experince with backend and apis. I'll wait on his approval, otherwise, with my doubts and nits covered LGTM. :smile:

File: server/src/configs/app.js
Comment (nihalanisumit) [inline server/src/configs/app.js]: remove empty line @piyushagwl007

File: server/src/routes/index.js
Comment (nihalanisumit) [inline server/src/routes/index.js:6]: @piyushagwl007 it's not a good idea to log like this when code goes for production. If you want to log for debug, try using this
https://expressjs.com/en/guide/debugging.html

File: server/src/routes/index.js
Comment (nihalanisumit) [reply server/src/routes/index.js:7]: @karansapolia read about middlewares in express.
but if I had to give you a gist I would say that in express middlewares are functions which are run through every request unless you return a response.
let's say you have a request coming and you want to pass it throught three functions (middlewares) each adding something to the request or modifying it, since you are not return anything from those  middlewares, it will never go out of first middleware so for this you use next(). After the end of every middlware you call next() this actually calls the next middlware unlessa middleware or function sends a response/ throws an error or returns something

File: server/src/configs/app.js
Comment (piyushagwl007) [reply server/src/configs/app.js:7]: It would be better if we pass some variables in the future and someone else is passing variables. It would become easy to resolve the conflict in that case.

File: server/src/configs/app.js
Comment (piyushagwl007) [reply server/src/configs/app.js:12]: Explained in above comment

File: server/src/controller/apis/user.js
Comment (piyushagwl007) [reply server/src/controller/apis/user.js:7]: This route will help us to check whether the user-specific APIs are working or not.
This will give us the status 200 if the users API are reachable.
eg:
```
'/health': To check the health of the most basic route.
'/api/v1/users/health': To check the health of the user route.

```

Now this thing would help us in debugging and writing tests as well.

File: server/src/routes/index.js
Comment (piyushagwl007) [reply server/src/routes/index.js:7]: This is a part of middleware.
You can read more about this [here](https://expressjs.com/en/guide/using-middleware.html).

File: server/src/routes/index.js
Comment (piyushagwl007) [reply server/src/routes/index.js:6]: Sure will use debug console

**Review by nihalanisumit — APPROVED** [review]
LGTM

**Pull request merged**
